mmeval/metrics/connectivity_error.py

```diff
@@ -0,0 +1,128 @@
+
+    Examples:
+
+        >>> from mmeval import ConnectivityError
+        >>> import numpy as np
+        >>>
+        >>> connectivity_error = ConnectivityError()
+        >>> pred_alpha = np.zeros((32, 32), dtype=np.uint8)
+        >>> gt_alpha = np.ones((32, 32), dtype=np.uint8) * 255
+        >>> trimap = np.zeros((32, 32), dtype=np.uint8)
+        >>> trimap[:16, :16] = 128
+        >>> trimap[16:, 16:] = 255
+        >>> connectivity_error(pred_alpha, gt_alpha, trimap)
+        {'ConnectivityError': ...}
+    """
+
+    def __init__(self,
+                 step: float = 0.1,
+                 norm_const: int = 1000,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.step = step
+        self.norm_const = norm_const
+
+        if cv2 is None:
+            raise ImportError(f'For availability of {self.__class__.__name__},'
+                              ' please install cv2 first.')
+
+    def add(self, pred_alphas: Sequence[np.ndarray], gt_alphas: Sequence[np.ndarray], trimaps: Sequence[np.ndarray]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add ConnectivityError score of batch to ``self._results``
+        Args:
+            pred_alphas (Sequence[np.ndarray]): Pred_alpha data of predictions.
+            ori_alphas (Sequence[np.ndarray]): Ori_alpha data of data_batch.
+            ori_trimaps (Sequence[np.ndarray]): Ori_trimap data of data_batch.
+        """
+
+        for pred_alpha, gt_alpha, trimap in zip(pred_alphas, gt_alphas,
+                                                trimaps):
+            assert pred_alpha.shape == gt_alpha.shape, 'The shape of ' \
+                '`pred_alpha` and `gt_alpha` should be the same, but got: ' \
+                f'{pred_alpha.shape} and {gt_alpha.shape}'
+
+            thresh_steps = np.arange(0, 1 + self.step, self.step)
+            round_down_map = -np.ones_like(gt_alpha)
+            for i in range(1, len(thresh_steps)):
+                gt_alpha_thresh = gt_alpha >= thresh_steps[i]
+                pred_alpha_thresh = pred_alpha >= thresh_steps[i]
+                intersection = gt_alpha_thresh & pred_alpha_thresh
+                intersection = intersection.astype(np.uint8)
+
+                # connected components
+                _, output, stats, _ = cv2.connectedComponentsWithStats(
+                    intersection, connectivity=4)
+                # start from 1 in dim 0 to exclude background
+                size = stats[1:, -1]
+
+                # largest connected component of the intersection
+                omega = np.zeros_like(gt_alpha)
+                if len(size) != 0:
+                    max_id = np.argmax(size)
+                    # plus one to include background
+                    omega[output == max_id + 1] = 1
+
+                mask = (round_down_map == -1) & (omega == 0)
+                round_down_map[mask] = thresh_steps[i - 1]
+            round_down_map[round_down_map == -1] = 1
```

Comment:
Suggest add a blank line here~
```suggestion
                round_down_map[mask] = thresh_steps[i - 1]

            round_down_map[round_down_map == -1] = 1
```